torch/_torch_docs.py

```diff
@@ -1242,17 +1242,23 @@ def merge_dicts(*dicts):
 is ``True`` then the returned tensor will require a gradient, and if :attr:`obj` is
 also a tensor with an autograd history then the returned tensor will have the same history.
 
-When :attr:`obj` is not a tensor, NumPy Array, or DLPack capsule but implements Python's
+When :attr:`obj` is not a tensor, NumPy array, or DLPack capsule but implements Python's
 buffer protocol then the buffer is interpreted as an array of bytes grouped according to
 the size of the datatype passed to the :attr:`dtype` keyword argument. (If no datatype is
 passed then the default floating point datatype is used, instead.) The returned tensor
 will have the specified datatype (or default floating point datatype if none is specified)
 and, by default, be on the CPU device and share memory with the buffer.
 
-When :attr:`obj` is none of the above but a scalar or sequence of scalars then the
+When :attr:`obj` is none of the above but a NumPy scalar, a scalar or a sequence of scalars then the
 returned tensor will, by default, infer its datatype from the scalar values, be on the
 CPU device, and not share its memory.
 
+.. note::
+
+    NumPy scalars also implement the buffer protocol. However, NumPy scalars are
+    prioritized over buffer protocols. In other words, if :attr:`obj` is a NumPy scalar,
+    it will not share memory, and its type will be inferred.
```

Comment:
This note is probably too prominent, and most users won't really care about the details here. The paragraph above suffices I think.